matrix_construction.py

```diff
@@ -101,14 +103,56 @@ def stack_matrices_in_folder(path: str):
     return doc
 
 
+def fill_half_matrix(matrix):
+    """
+    Fills out the other half of symmetric matrix, that has only been half filled out to optimize.
+    :param matrix: sparse matrix
+    :return: filled out sparse matrix
+    """
+    # if the matrix size is uneven, cut to the smallest even size.
+    size = min(matrix.shape[0], matrix.shape[1])
+    matrix = matrix[:size, :size]
+    # add transposed matrix and make diagonal 0
+    matrix = matrix.todense()
+    matrix = matrix + matrix.T
+    np.fill_diagonal(matrix, 0)
+    return sp.csr_matrix(matrix)
+
+
+def matrix_connection_check(adj_matrix) -> bool:
+    """
+    Checks if a graph is fully connected repeatedly checking unvisited neighbors.
```

Comment:
Som jeg forstår koden, tjekker den ikke bare at grafen er connected, ikke fully connected (hvilket er det samme som en complete graph)?

Reply:
Den checker om man kan komme fra enhver node til enhver anden node vel ?

Reply:
Ja, altså om der er en path fra alle noder til alle noder. Det hedder bare at den er connected.
```suggestion
    Checks if a graph is connected, repeatedly checking unvisited neighbors.
```